src/server/services/wms/qgswmslayerinfos.h

```diff
@@ -0,0 +1,152 @@
+/***************************************************************************
+                              qgswmslayerinfos.h
+
+  Layer's information
+  ------------------------------------
+  begin                : September 26 , 2022
+  copyright            : (C) 2022 by René-Luc D'Hont and David Marteau
+  email                : rldhont at 3liz doc com
+         dmarteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSWMSLAYERINFOS_H
+#define QGSWMSLAYERINFOS_H
+
+#include "qgsrectangle.h"
+#include "qgsmaplayerserverproperties.h"
+
+class QgsServerInterface;
+class QgsProject;
+
+/**
+ * \ingroup server
+ * \brief WMS Layer infos
+ */
+
+namespace QgsWms
+{
+  struct layerInfos
```

Comment:
Shouldn't this be `LayerInfos` ? We don't use namespaces so much so I'm not sure.

Reply:
In `src/server/services/wmts/qgswmtsutils.h`, we defined struct with a lower first letter. For exemple `tileMatrixInfo`, `tileMatrixDef`, `tileMatrixSetDef`, `tileMatrixLimitDef`.

@dmarteau is the concention for struct is the same as for namespace ?

Reply:
Don't take the server code base as a good example of naming conventions, there are so many odd things in there.
We should try to fix them all but more important is to not introduce new ones.

Also, I don't think we have any coding guidelines for namespaces because AFAIK they are only used in the server code.

In general, class names and struct names and enum names should be all uppercased.

Reply:
Ok fixed